frontend/app/components/error-summary.tsx

```diff
@@ -1,48 +1,248 @@
+import { AnchorLink } from '~/components/anchor-link';
+import { cn } from '~/utils/tailwind-utils';
+
+/**
+ * Represents an item in the error summary, containing an error message
+ * and the associated field ID where the error occurred.
+ */
+export interface ErrorSummaryItem {
+  /** The error message to display. */
+  errorMessage: string;
+  /** The ID of the field associated with the error. */
+  fieldId: string;
+}
 
-interface ErrorSummaryProps {
-  errors?: NestedErrors;
-  full?: boolean;
+/**
+ * Props for the ErrorSummary component.
+ */
+export interface ErrorSummaryProps extends OmitStrict<ComponentPropsWithoutRef<'section'>, 'role' | 'tabIndex'> {
+  /** A list of error items to display. */
+  errors: readonly ErrorSummaryItem[];
 }
 
-export function ErrorSummary({ errors, full }: ErrorSummaryProps) {
+/**
+ * ErrorSummary component displays a list of error messages with associated
+ * field IDs and provides anchor links for users to quickly navigate to
+ * specific form fields that caused the errors.
+ *
+ * - When there are errors, the component scrolls the section into view
+ *   and focuses on it to ensure visibility.
+ * - Provides an accessible, styled list of errors with links to the form fields.
+ *
+ * @returns The rendered component or `false` if there are no errors.
+ */
+export function ErrorSummary({ className, errors, ...rest }: ErrorSummaryProps): JSX.Element | false {
   const { t } = useTranslation(['gcweb']);
+  const sectionRef = useRef<HTMLElement>(null);
+
+  // Scroll and focus on the error summary when errors are updated.
+  useEffect(() => {
+    if (errors.length > 0 && sectionRef.current) {
+      sectionRef.current.scrollIntoView({ behavior: 'smooth' });
+      sectionRef.current.focus();
+    }
+  }, [errors]);
+
+  // If no errors, return false (no rendering).
+  if (errors.length === 0) {
+    return false;
+  }
+
+  return (
+    <section
+      ref={sectionRef}
+      className={cn('my-5 border-4 border-red-600 p-4', className)}
+      {...rest}
+      tabIndex={-1}
+      role="alert"
+    >
+      <h2 className="font-lato text-lg font-semibold">{t('gcweb:error-summary.header', { count: errors.length })}</h2>
+      <ol className="mt-1.5 list-decimal space-y-2 pl-7">
+        {errors.map(({ fieldId, errorMessage }) => (
+          <li key={`${fieldId}-${errorMessage}`}>
+            <AnchorLink className="text-red-700 underline hover:decoration-2 focus:decoration-2" anchorElementId={fieldId}>
+              {errorMessage}
+            </AnchorLink>
+          </li>
+        ))}
+      </ol>
+    </section>
+  );
+}
+
+/**
+ * Props for the `ActionDataErrorSummary` component.
+ */
+type ActionDataErrorSummaryProps = ComponentPropsWithoutRef<'div'> & {
+  /**
+   * The action data from a React Router Form or Fetcher.
+   * This triggers error collection when form or Fetcher data changes.
+   */
+  actionData: unknown | undefined;
```

Comment:
Any reason this isn't `actionData?: unknown`?

Reply:
This component doesn't need to know what type `actionData` is. It only have to know if the `actionData` has changed.